frontend/app/views/fragments/form/submit.scala.html

```diff
@@ -37,7 +39,14 @@
             <noscript>
                 <a href="http://www.enable-javascript.com/" target="_blank" class="action is-disabled">Enable javascript to proceed</a>.
             </noscript>
-            <button type="button" class="action js-submit-input u-hidden-non-js">
+            <button type="button" class="action js-submit-input u-hidden-non-js"
+                    @if(dataCategory != None && dataLabel != None) {
```

Comment:
Originally I tried to implemented with a for expr (for-yield) but that kind of expression is not supported by the scala template engine. The way that you suggested above is not supported either. 

The only way to do this using a for expr is: 

```
@for(
  label <- dataLabel
) {
  data-metric-label=label
}
@for(
  category <- dataCategory
) {
  data-metric-category=category
}
```